scipy/stats/stats.py

```diff
@@ -237,6 +237,28 @@ def _chk2_asarray(a, b, axis):
     return a, b, outaxis
 
 
+def _contains_nan(a, nan_policy='propagate'):
+    if nan_policy not in ('propagate', 'raise', 'omit'):
+        raise ValueError("nan_policy must be either 'propagate', 'raise', or "
+                         "'ignore'")
+    try:
+        # Calling np.sum to avoid creating a huge array into memory
+        # e.g. np.isnan(a).any()
+        contains_nan = np.isnan(np.sum(a))
+    except TypeError:
+        # If the check cannot be properly performed we fallback to omiting
+        # nan values and raising a warning.
```

Comment:
The only thing that now triggers this is ``mode`` with an array of strings right? Might want to mention that, otherwise it's a bit unclear where the ``TypeError`` could come from.